rasa/core/nlg/response.py

```diff
@@ -25,28 +26,55 @@ def __init__(self, responses: Dict[Text, List[Dict[Text, Any]]]) -> None:
         """
         self.responses = responses
 
+    def _matches_filled_slots(
+        self, filled_slots: Dict[Text, Any], response: Dict[Text, Any],
+    ) -> bool:
+        """Checks if the conditional response variation matches the filled slots."""
+        constraints = response.get(CONDITION)
```

Comment:
optional: maybe the name `RESPONSE_CONDITION` would make the usage of the constant a bit clearer